spawn/spawn.go

```diff
@@ -160,3 +180,106 @@ func CreateHost(so Options) (*host.Host, error) {
 	}
 	return intentHost, errors.WithStack(err)
 }
+
+func SetHostRDPPassword(ctx context.Context, host *host.Host, password string) error {
+	pwdUpdateCmd, err := constructPwdUpdateCommand(evergreen.GetEnvironment().Settings(), host, password)
+	if err != nil {
+		return errors.Wrap(err, "Error constructing host RDP password")
+	}
+
+	// update RDP and sshd password
+	if err = pwdUpdateCmd.Run(ctx); err != nil {
+		return errors.Wrap(err, "Error updating host RDP password")
+	}
+	return nil
+}
+
+// constructPwdUpdateCommand returns a RemoteCommand struct used to
+// set the RDP password on a remote windows machine.
+func constructPwdUpdateCommand(settings *evergreen.Settings, hostObj *host.Host,
+	password string) (*subprocess.RemoteCommand, error) {
+
+	cloudHost, err := providers.GetCloudHost(hostObj, settings)
+	if err != nil {
+		return nil, err
+	}
+
+	hostInfo, err := util.ParseSSHInfo(hostObj.Host)
+	if err != nil {
+		return nil, err
+	}
+
+	sshOptions, err := cloudHost.GetSSHOptions()
+	if err != nil {
+		return nil, err
+	}
+
+	stderr := send.MakeWriterSender(grip.GetSender(), level.Error)
+	defer stderr.Close()
+	stdout := send.MakeWriterSender(grip.GetSender(), level.Info)
+	defer stdout.Close()
+
+	updatePwdCmd := fmt.Sprintf("net user %v %v && sc config "+
+		"sshd obj= '.\\%v' password= \"%v\"", hostObj.User, password,
+		hostObj.User, password)
+
+	// construct the required termination command
+	remoteCommand := &subprocess.RemoteCommand{
+		CmdString:       updatePwdCmd,
+		Stdout:          stdout,
+		Stderr:          stderr,
+		LoggingDisabled: true,
+		RemoteHostName:  hostInfo.Hostname,
+		User:            hostObj.User,
+		Options:         append([]string{"-p", hostInfo.Port}, sshOptions...),
+		Background:      false,
+	}
+	return remoteCommand, nil
+}
+
+func TerminateHost(host *host.Host, settings *evergreen.Settings) error {
+	if host.Status == evergreen.HostTerminated {
+		return errors.New("Host is already terminated")
+	}
+	if host.Status == evergreen.HostUninitialized {
+		return host.SetTerminated()
+	}
+	cloudHost, err := providers.GetCloudHost(host, settings)
+	if err != nil {
+		return err
+	}
+	if err = cloudHost.TerminateInstance(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func MakeExtendedHostExpiration(host *host.Host, addHours int) (time.Time, error) {
+	addHoursDuration := time.Duration(addHours) * time.Hour
+	newExp := host.ExpirationTime.Add(addHoursDuration)
+	hoursToExtendBy := newExp.Sub(time.Now()).Hours() //nolint
+	if hoursToExtendBy > MaxExpirationDurationHours {
+		return time.Time{}, errors.Errorf("Can not extend host '%s' expiration by '%d' hours. Maximum extension is limited to %d hours", host.Id, addHours, MaxExpirationDurationHours)
+	}
+
+	return newExp, nil
+}
+
+func ValidateRDPPassword(password string) bool {
+	// Golang regex doesn't support lookarounds, so we can't use
+	// the regex as found in public/static/js/directives/directives.spawn.js
```

Comment:
Should I change the UI to use the same regexes as golang?

Reply:
if it's easy?